### ISSUE TITLE
(new pdf viewer) No escape (literally) from fullscreen PDF in Chrome and Edge

### ISSUE THREAD

**Issue by andrewjbtw** [sul-dlss/sul-embed#2162]
In Chrome, if you full screen a PDF in the browser viewer, you can get stuck without being able to escape from full screen.

To reproduce:

- Go to a PDF in stage where the browser viewer is deployed: https://sul-purl-stage.stanford.edu/bg077wm0255
- Make embed full screen
- Click on the PDF display: just click within the PDF viewer window, highlight text, etc.
- Press escape and see that you can't get out of the full screen

Additional information:

- All my testing is on a Mac (it's what I have)
- I have not been able to reproduce this in Firefox or Safari, just Chrome and Edge
- If you don't click within the PDF window, you can still use escape to leave full screen. The issue only comes up if you interact with the PDF

**Comment by jmartin-sul** [comment]
After much (unsuccessful) testing of possible fixes for the issue, and much discussion, we decided to just disable the full screen button for browsers that we know are based on Chromium (Chrome and Edge, at the moment).  @andrewjbtw is ok with this because in his comparison of our PDF viewer with the PDF viewers of other sites and web apps, he noticed that not all the other viewers offered a full screen button.  Additionally, any PDF that is viewable in the sul-embed PDF viewer is downloadable, so a Chromium user who really wants to full screen the PDF can just download it and view their local copy in the PDF viewer of their choice.

Also worth noting: this is a bug that others have been running into for at least a few years, and there seems to be no indication of when it'll be fixed: https://issues.chromium.org/issues/40131988

A small bit of additional nuance: if you full screen the PDF in Chrome or Edge, the escape key does actually work until (as Andrew describes) you click into the PDF, or click away from the browser and then back into the PDF (as you might if you have a second monitor).

Things we tried that didn't fix the escape key bug:
* constantly returning focus to an element in the DOM outside of the native PDF viewer's `object` tag, a la the suggestion in this stack overflow post (which fixed the escape key issue, but broke keyboard navigation within the PDF, thus breaking accessibility for the viewer): https://stackoverflow.com/questions/58702747/window-events-with-pdf-document-via-chrome
* trying to put focus on the `object` tag from within the [fscreen click listener that makes the PDF viewer go full screen](https://github.com/sul-dlss/sul-embed/blob/011a7a3d0c62c44180520386e8f89c79e079c6dc/app/javascript/src/modules/fullscreen.js#L10-L12) (fscreen is the JS library we use for cross browser full screen behavior so that we don't have to use differing browser APIs directly).
* catching key events using a Stimulus JS controller on the `div` around the PDF viewer `object` tag.
* attaching a key listener to something within the native PDF viewer's DOM (in Chromium based browsers the PDF viewer was in [a separate shadow DOM](https://stackoverflow.com/questions/55704303/event-listener-outside-of-shadow-dom-wont-bind-to-elements-inside-of-shadow-dom), and we couldn't find a way to programmatically get access to that shadow DOM; additionally, even when I tried manually adding listeners to PDF viewer elements in the shadow DOM using the Chrome developer console, events were only detected for some parts of the viewer and not others, e.g. events were detected in the page selection sidebar but not from the page elements themselves, which still left large areas on which the user could click and break escape key behavior)

We also briefly considered popping up a message to Chromium users telling them that the escape key wouldn't work for exiting full screen, but contradicting the browser and telling users that it's lying felt like worse UX than just removing the full screen button for users for whom it wouldn't work consistently.
